packages/jest/src/index.tsx

```diff
@@ -1,8 +1,15 @@
 declare global {
   namespace jest {
     interface Matchers<R> {
-      toHaveCompiledCss(properties: { [key: string]: string }): R;
-      toHaveCompiledCss(property: string, value: string): R;
+      toHaveCompiledCss(
+        properties: { [key: string]: string },
+        matchFilter?: Partial<Record<'state' | 'media', string>>
+      ): R;
+      toHaveCompiledCss(
+        property: string,
+        value: string,
+        matchFilter?: Partial<Record<'state' | 'media', string>>
```

Comment:
can bring in `Pseudos` to make it a bit easier and then fallback to `string`?

```
import * as CSS from 'csstype';

type State = Pseudos;
```

should we rename `state` to something that encompasses other groups pseudo selectors as well? what's your thoughts

Reply:
Wow, state to 'selector' maybe? I'll try this and update

Reply:
@madou if we just use the value as Pseudos, TS will automatically suggest and add `:`, but then `> :first-child` this won't be a valid one.

Is there a way I can preserve the intellisense as well as adding string?

`Pseudos | string` is just `string`

Reply:
let me check it out locally and look

Reply:
found an issue https://github.com/microsoft/TypeScript/issues/29729

Reply:
```
import { Pseudos } from 'csstype';

/**
 * {} is used to enabled the compiler to not reduce Target down to a string.
 * This keeps both Pseduos intellisense working as well as then allowing us to define anything.
 * See: https://github.com/microsoft/TypeScript/issues/29729#issuecomment-460346421
 */
type AnyTarget = string & { _?: never };
type Target = Pseudos | AnyTarget;

export interface MatchFilter {
  target?: Target;
  media?: string;
}
```

this fixes it